### ISSUE TITLE
Reverse Compass

### ISSUE THREAD

**Issue by chkwan** [gartnera/headunit#148]
When I change the "reverseGPS" flag to true wihtin  .\config\androidautowifi\headunit.json, AA will fail to launch and will crash the CMU. When the flag is false, the application boots fine but my GMaps navigation is reversed.

**Comment by Trevelopment** [comment]
The code for that code would be right here https://github.com/gartnera/headunit/blob/61ee618f2a7a4ce8c8d07ef7a2690a3a81724ca4/mazda/main.cpp#L125
there is probably something wrong there

**Comment by Trevelopment** [comment]
You can also just remove your Nav SD card to fix this issue.

**Comment by chkwan** [comment]
I will try to remove the NAV SD but this wasn't an issue in 1.09. GMaps and Waze both faced the correct direction with the NAV SD card inserted.

**Comment by Trevelopment** [comment]
Really? what FW version are you using?

**Comment by chkwan** [comment]
CMU_SW_VER = 55.00.753A-NA

With the newest 1.10+ version and "reverseGPS" flag set to true, AA does not even boot and just crashes. So I am unable to test whether this flag fixes the compass or not.

Not sure if this helps but to clarify, by crashing, I mean it never gets off of the Mazda Application screen and none of the buttons are responsible. The radio still keeps on playing though.

**Comment by Trevelopment** [comment]
Yea I believe it crashes the strange thing about this is that I have the same FW version and am not experiencing the same issue.

**Comment by chkwan** [comment]
I have tried different USB drives with reformat to FAT32 each time. Unfortunately, I'm not a programmer my ability to self-diagnose is limited. Any other thoughts or things I should try?

I MZD-AIO v2.8.3 to build the files and Notepad++ to edit the json file.

**Comment by Trevelopment** [comment]
So I did some testing myself did you try not choosing wifi and were you checking the reverse compass option in AIO and setting the flag or just one or the other?  because when I dont check wifi support but check reverse compass AND set the flag in headunit.json to true then the compass is normal until I insert my NAV SD card then it reverses the heading. Note that I had no issue before so the option is giving me the issue.

**Comment by Peck07** [comment]
It was a long time ago... but I think I've just put that reverse compass fix under a flag, as not everyone needs it. But if the fix checks for SD card, it checks still... I was considering to remove that SD card check, but as I didn't have this issue, I've just kept the original code from here:
https://github.com/gartnera/headunit/pull/123

**Comment by Trevelopment** [comment]
Well hey it works so might as well keep it in there it doesnt even check for the sd card if the flag is not set.  I just made it an option you can choose in AIO without editing the json file to try and make it easier but now I want to make sure its all working as expected so I can get another release out even if I cant figure out how to get headunit to run on v70 with the stock USB hub yet.  But really I keep my Nav SD card popped out while using AA anyway because it causes me performance issues but I don''t have that reversed compass issue normally either, this was actually the first time I experienced it after setting the flag.

**Comment by Peck07** [comment]
Yeah, I understand, but apparently it's causing some issues for someone :(
But to be honest, I don't think this code caused the reverse compass issue between 1.09 and 1.10+ for chkwan (that's another mystery).
For the reverseGPS flag: maybe notepad++ screws something up in the file (I'm not blaming, just guessing)

**Comment by Trevelopment** [comment]
Thats not what hes saying, I am the one saying if you dont have a reversed compass then choosing that option will reverse your compass but hes saying he has the reversed compass so he needs to use your fix as the solution to his issue.

**Comment by Peck07** [comment]
I know, but turning the reverseGPS on doesn't work for some reason for chkwan, as AA crashes :)

**Comment by Trevelopment** [comment]
I'm saying it does work

**Comment by Trevelopment** [comment]
I'm in my car right now using AA mirror typing on my car screen on my phone anyways I just had that flag set and it would reverse my heading when my nav SD card is in. I just sent the flag to false right now and now when I put my nav SD card in it doesn't change the heading at all.

**Comment by chkwan** [comment]
Hey guys, thank you for your help. What I tried this time was to build the clean flash drive using MZD AIO. I selected enable WIFI at the the top and the Android Auto Headunit App setting only. Within Android Auto settings screen, I turned off WiFi Support and enabled reverse compass. Then I compiled the flash drive. When I checked the JSON file, "reverseGPS" was still still set to false so I changed it to ture.

I then flashed this to the car and GMaps seem to be working fine now. Thanks!
